build_tools/docker/mmperf/Dockerfile

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# An image for running SHARK tank: https://github.com/nod-ai/SHARK.
+
+FROM ubuntu:22.04
+
+######## Basic ########
+RUN apt-get update \
+  && apt-get install -y \
+    git \
+    wget \
+    cmake \
+    curl \
+    ninja-build
+##############
+
+######## Clang/LLVM ########
+RUN apt-get update \
+  && apt-get install -y \
+    llvm-14 \
+    llvm-14-dev \
+    clang-14 \
+    clang-tools-14 \
+    libclang-common-14-dev \
+    libclang-14-dev \
+    libclang1-14 \
+    clang-format-14 \
+    clangd-14 \
+    clang-tidy-14 \
+    lldb-14 \
+    lld-14 \
+    libmlir-14-dev \
+    mlir-14-tools \
```

Comment:
MLIR is used here: https://github.com/mmperf/mmperf/blob/7651ce05e28c7e81f8a0e6867177b4d2894b6b95/matmul/matmul-compile/CMakeLists.txt#L20.

Since this was not included in the mmperf submodules where a specific version is needed, I assumed that a prebuilt would suffice.

Reply:
Ah, that's using the upstream `MLIRExecutionEngine` code. Maybe that's self-contained and can use whatever version is published for the distro. I certainly wouldn't trust something like that for any downstream code - the code in MLIR upstream changes far too regularly.

Reply:
Yeah mmperf also runs the MLIR sandbox but for that it depends on an llvm submodule. IREE also uses its own version of LLVM/MLIR.